app/player/knight.py

```diff
@@ -0,0 +1,42 @@
+from __future__ import annotations
+
+
+class Knight:
+
+    def __init__(self,
+                 name: str,
+                 power: int,
+                 hp: int,
+                 armour: list,
+                 weapon: dict,
+                 potion: dict
+                 ) -> None:
+
+        self.name = name
+        self.power = power
+        self.hp = hp
+        self.protection = 0
+
+        for item in armour:
+            self.protection += item["protection"]
+
+        self.power += weapon["power"]
+
+        potion_item = ["power", "protection", "hp"]
+
+        if potion is not None:

```

Comment:
It is always false. It your `__init__` you say that potion always dict
Update in init type hint